client/dashboard/app/queries/domain-whois.ts

```diff
@@ -0,0 +1,34 @@
+import { mutationOptions, queryOptions } from '@tanstack/react-query';
+import {
+	fetchDomainWhois,
+	fetchDomainWhoisValidate,
+	updateDomainWhois,
+} from '../../data/domain-whois';
+import { queryClient } from '../query-client';
+import type { DomainContactDetails } from '../../domains/contact-details/types';
+
+export const domainWhoisQuery = ( domainName: string ) =>
+	queryOptions( {
+		queryKey: [ 'domains', domainName, 'whois' ],
+		queryFn: () => fetchDomainWhois( domainName ),
+	} );
+
+export const domainWhoisValidateQuery = ( domainName: string, contactInformation: any ) =>
+	queryOptions( {
+		queryKey: [ 'domains', domainName, 'whois', 'validate' ],
+		queryFn: () => fetchDomainWhoisValidate( domainName, contactInformation ),
+	} );
+
+export const updateDomainWhoisMutation = ( domainName: string ) =>
```

Comment:
[nit] if the mutation is literally an "update", we tend not to include it in the mutation at all, to make it symmetric with the query


```suggestion
export const domainWhoisMutation = ( domainName: string ) =>
```